hikari/internal/collections.py

```diff
@@ -234,24 +254,45 @@ class LimitedCapacityCacheMap(ExtendedMutableMapping[KeyT, ValueT]):
         it starts removing the oldest entries.
     source : typing.Optional[typing.Dict[KeyT, ValueT]]
         A source dictionary of keys to values to create this from.
+
+    Other Parameters
+    ----------------
+    on_expire : typing.Optional[typing.Callable[[ValueT], None]]
```

Comment:
probably add a note about how this will be called after it's been removed from the mapping